src/platform/Platform.c

```diff
@@ -0,0 +1,680 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <Platform.h>
+#include <Module.h>
+#include <Log.h>
+
+#include <dirent.h>
+#include <stdbool.h>
+#include <string.h>
+#include <time.h>
+
+#include <parson.h>
+
+#define MODULE_EXT ".so"
+
+#define AZURE_OSCONFIG "Azure OSConfig"
+
+#define MODEL_VERSION "ModelVersion"
+#define REPORTED "Reported"
+#define COMPONENT_NAME "ComponentName"
+#define OBJECT_NAME "ObjectName"
+
+#define UUID_LENGTH 36
+
+typedef struct MODULE_SESSION
+{
+    MODULE* module;
+    MMI_HANDLE handle;
+
+    struct MODULE_SESSION* next;
+} MODULE_SESSION;
+
+typedef struct SESSION
+{
+    char* uuid;
+    char* client;
+    MODULE_SESSION* modules;
+
+    struct SESSION* next;
+} SESSION;
+
+typedef struct REPORTED_OBJECT
+{
+    const char* component;
+    const char* object;
+} REPORTED_OBJECT;
+
+static SESSION* g_sessions = NULL;
+static MODULE* g_modules = NULL;
+static REPORTED_OBJECT* g_reportedObjects = NULL;
+static int g_numReportedObjects = 0;
+
+void LoadModules(const char* directory, const char* configJson)
+{
+    MODULE* module = NULL;
+    DIR* dir = NULL;
+    struct dirent* entry = NULL;
+    char* path = NULL;
+    char* client = NULL;
+    int version = 0;
+    const char* component = NULL;
+    const char* object = NULL;
+    int reportedCount = 0;
+    JSON_Value* config = NULL;
+    JSON_Object* configObject = NULL;
+    JSON_Array* reportedArray = NULL;
+    JSON_Object* reportedObject = NULL;
+    REPORTED_OBJECT* reported = NULL;
+
+    if (g_modules != NULL)
+    {
+        return;
+    }
+
```

Comment:
Why not treat this as reaload? This right now does not seem OK. If keeping, trace this. We do not want this function to be called all the time.